hdf5.spec

```diff
@@ -13,18 +14,16 @@
 # You need to recompile all users of HDF5 for each version change
 Name: hdf5
 Version: %{major}.%{minor}
-Release: 4%{?dist}
+Release: 5%{hdf5_sha}%{?dist}
 Summary: A general purpose library and file format for storing scientific data
 License: BSD
 URL: https://portal.hdfgroup.org/display/HDF5/HDF5
 
-Source0: http://www.hdfgroup.org/ftp/HDF5/releases/hdf5-%{major}/hdf5-%{version}%{?snaprel}/src/hdf5-%{version}%{?snaprel}.tar.bz2
+Source0: https://github.com/HDFGroup/hdf5/archive/%{hdf5_commit}.tar.gz
```

Comment:
I much prefer this to be an actual upstream release and then add a `Patch:` to bring it up to master.  It's more transparent about what the resulting code is that way.

The upstream release should be the closest release to the commit from master being applied.